### PR TITLE
Feature/better imm filter

### LINKED ISSUE #9: [TASK] IMM Filter
### Description of task

IMM Filter

### Suggested Workflow

_No response_

### Specifications

_No response_

### Contacts

_No response_

### Code Quality

- [ ] Every function in header files are documented (inputs/returns/exceptions)
- [ ] The project has automated tests that cover MOST of the functions and branches in functions (pytest/gtest)
- [ ] The code is documented on the wiki (provide link)

### PR DESCRIPTION
**Pull request by EirikKolas**
IMM filter with support for models of different state-sizes

I used a lot of variadic templates and integer sequences to make the user interface easy to use but it does make the code a bit hard to understand. Please comment if something is unclear

### CODE REVIEW

File: vortex-filtering/include/vortex_filtering/filters/README.md [hunk 1]
```diff
@@ -1,38 +1,228 @@
 # Filters
-This folder contains various filters
+This folder contains various filters for state estimation. Th
```
Comment (alekskl01) [inline vortex-filtering/include/vortex_filtering/filters/README.md]: Th?

File: vortex-filtering/include/vortex_filtering/filters/README.md [hunk 2]
```diff
@@ -1,38 +1,228 @@
...
 
 All classes and functions are under the namespace `vortex::filters`.
 
-## EKF
-This class represents an [Extended Kalman Filter](https://en.wikipedia.org/wiki/Extended_Kalman_filter). It is a template class with parameters `DynamicModelT` and `SensorModelT` for the dynamic model and sensor model respectively. The models have to be derived from `vortex::models::DynamicModelLTV` and `vortex::models::SensorModelLTV`. All methods are static, so there is no need to create an instance of this class.
+Here is a great intro to Kalman filters in general: [Kalman and Bayesian Filters in Python](
```
Comment (alekskl01) [inline vortex-filtering/include/vortex_filtering/filters/README.md]: where link?

Comment (EirikKolas) [reply vortex-filtering/include/vortex_filtering/filters/README.md]: here link

File: vortex-filtering/include/vortex_filtering/filters/README.md (hunk 1 above)
Comment (EirikKolas) [reply vortex-filtering/include/vortex_filtering/filters/README.md]: The filters are based on the Kalman filter and its extensions, and are designed to work with the models in the `vortex::models` namespace.

File: vortex-filtering/README.md [hunk 3]
```diff
@@ -6,4 +6,72 @@ Contains the models used in the filters. The models are implemented as classes t
...
\ No newline at end of file
+
+```mermaid
+classDiagram
+
+    DynamicModel <|-- DynamicModelLTV
+    SensorModel <|-- SensorModelLTV
+    DynamicModelLTV <|-- ConstantVelocity
+    DynamicModelLTV <|-- ConstantAcceleration
+    DynamicModelLTV <|-- CoordinatedTurn
```
Comment (alekskl01) [inline vortex-filtering/README.md:17]: The direction of the arrows seem a bit misleading. If its inheretance, should they not point in the opposite direction? If not, what do they mean?

File: vortex-filtering/include/vortex_filtering/probability/multi_var_gauss.hpp [hunk 4]
```diff
@@ -93,15 +93,39 @@ template <size_t n_dims> class MultiVarGauss {
...
 template <int n_dims> using Gauss = MultiVarGauss<n_dims>;
-using Gauss2d                     = Gauss<2>;
-using Gauss3d                     = Gauss<3>;
-using Gauss4d                     = Gauss<4>;
+
+using Gauss2d = Gauss<2>;
+using Gauss3d = Gauss<3>;
+using Gauss4d = Gauss<4>;
+using Gauss5d = Gauss<5>;
+using Gauss6d = Gauss<6>;
```
Comment (alekskl01) [inline vortex-filtering/include/vortex_filtering/probability/multi_var_gauss.hpp:129]: Not sure if there are any better alternatives, but this seems like a messy way of doing this.

File: vortex-filtering/include/vortex_filtering/models/README.md [hunk 5]
```diff
@@ -22,7 +31,7 @@ Dynamic model interface for other classes to derive from. The [UKF class](../fil
 - Sampling Methods: Provides methods for sampling from the discrete time dynamics.
 
 #### DynamicModelLTV
-Dynamic model interface for other classes to derive from. The [EKF class](../filters/README.md#ekf) works on models derived from this class.
+Dynamic model interface for other classes to derive from. The [EKF](../filters/README.md#ekf) (and UKF) works on models derived from this class.
 
 This interface inherits from the `DynamicModel` interface and defines the dynamics of the system as a linear time varying system. The virtual method `f_d` from the `DynamicModel` interface is implemented as a linear time varying system on the form 
 $$
```
Comment (alekskl01) [inline vortex-filtering/include/vortex_filtering/models/README.md:38]: It does not display correctly on github

**Review by alekskl01 — APPROVED** [review]

**Pull request merged**